src/themes/default/components/core/blocks/Microcart/EditMode.vue

```diff
@@ -30,6 +31,17 @@ export default {
       }
       this.product.qty = this.getEditingQty
       this.$store.dispatch('cart/configureItem', { product: this.product, configuration })
+    },
+    getEditedProduct (filter = {}) {
+      const selectedFilters = {...this.getSelectedOptions, [filter.type]: filter}
+      const sizeId = get(selectedFilters, 'size.id', '')
+      const colorId = get(selectedFilters, 'color.id', '')
+      const children = this.product.configurable_children || []
+
+      return children.find((child) =>
+        Number(child.color) === Number(colorId) &&
```

Comment:
This assumption is wrong; the configurable attribues can be different than size and color (they're defined in `product.configurable_attributes`)

Reply:
now you can edit only `size` and `color` it's restricted in template
`<div class="flex flex-wrap pt5" v-if="option.label == 'Color' && editMode">`
`<div class="flex flex-wrap pt5" v-else-if="option.label == 'Size' && editMode">`
So if there are different options (not editable) then it will fallback to `this.product`